src/query/storages/parquet/src/parquet_table/table.rs

```diff
@@ -21,32 +21,55 @@ use chrono::Utc;
 use common_arrow::arrow::datatypes::DataType as ArrowDataType;
 use common_arrow::arrow::datatypes::Field as ArrowField;
 use common_arrow::arrow::datatypes::Schema as ArrowSchema;
+use common_catalog::plan::DataSourceInfo;
 use common_catalog::plan::DataSourcePlan;
+use common_catalog::plan::ParquetReadOptions;
+use common_catalog::plan::ParquetTableInfo;
 use common_catalog::plan::PartStatistics;
 use common_catalog::plan::Partitions;
 use common_catalog::plan::PushDownInfo;
 use common_catalog::table::Table;
-use common_catalog::table_args::TableArgs;
 use common_catalog::table_context::TableContext;
 use common_exception::Result;
-use common_expression::Scalar;
 use common_expression::TableSchema;
 use common_meta_app::schema::TableIdent;
 use common_meta_app::schema::TableInfo;
 use common_meta_app::schema::TableMeta;
+use common_meta_types::UserStageInfo;
 use common_pipeline_core::Pipeline;
 use opendal::Operator;
 
-use crate::ReadOptions;
-
 pub struct ParquetTable {
-    pub(super) table_args: TableArgs,
-
     pub(super) file_locations: Vec<String>,
     pub(super) table_info: TableInfo,
     pub(super) arrow_schema: ArrowSchema,
     pub(super) operator: Operator,
-    pub(super) read_options: ReadOptions,
+    pub(super) read_options: ParquetReadOptions,
+    pub(super) stage_info: Option<UserStageInfo>,
```

Comment:
Why should `ParquetTable` hold `stage_info` ?

Reply:
we need it to reconstruct operator